pcrmc/pcrmc.py

```diff
@@ -60,12 +60,34 @@ def add(self,
         read = self._db_handler.read_contacts()
         if read.error != SUCCESS:
             return ContacterResponse(contact, read.error)
-        # TODO: this breaks/duplicates an ID if we remove and then add a contact # noqa: E501
-        # contact["ID"] = len(read.contact_list)
+
+        contact["ID"] = \
+            self._db_handler.get_new_contact_id(config.CONFIG_FILE_PATH)
```

Comment:
Hmm I think we should put this logic into the model layer, so database.py. The Controller shouldn't be concerned with ID generation, but should assume that the model takes care of it correctly. Compare to SQL, where the "unique autoincrement int" field is completely taken care of by sql, so the model. The controller just gives it the data and lets sql autogenerate the ID.

Reply:
i also thought about that but it would require more extensive changes in database.py. The DatabaseHandler has no idea what a contact is, yet.

Reply:
Let's change database.py then. One thought I had was to add more specific write functions, like write_all() (this is what we currently have) and write_contact() which appends a single given contact passed in as an object. The Controller would still be responsible to assemble the object, then the database writes it (and deals with stuff like the ID). This way, the database can then return the object that it wrote (with ID and all) back to the controller and controller to cli.

Reply:
yeah i think that is ultimately a good idea. swapping the model later would be easier as well